lib/Admin_Handler.cpp

```diff
@@ -322,6 +342,38 @@ bool FlushCommandWrapper(S* sess, const string& modname, char *query_no_space, i
 	return false;
 }
 
+std::tuple<bool, enum SERVER_TYPE, time_t>&& parse_command_purge_query_digests(char *query, int query_len) {
+	auto ret = new std::tuple<bool, enum SERVER_TYPE, time_t>(false, SERVER_TYPE_MYSQL, 0);
+
+	const char *prefix = match_command_prefix(CMD_PREFIX_PURGE_QUERY_DIGESTS, query, query_len);
+	if (prefix) {
+		// match_found
+		std::get<0>(*ret) = true;
+
+		// server_type
+		if (strstr(prefix, "_pgsql_") != nullptr) {
+			std::get<1>(*ret) = SERVER_TYPE_PGSQL;
+		}
+
+		// parse timestamp
+		char *timestamp = strdup(query + strlen(prefix));
+		int last_seen = atoi(trim_spaces_in_place(timestamp));
+
+		// match_not_found
+		// treat empty or non-positive timestamp as invalid command
+		if (last_seen <= 0) {
+			std::get<0>(*ret) = false;
+		}
+
+		// set last_seen
+		std::get<2>(*ret) = realtime_to_monotonic_time(last_seen);
```

Comment:
Suggestion: Instead of relying on tuple indexing with `std::get`, use individual named variables that represent the meaningful names for each of the tuple's elements. This can improve the code's clarity and make it easier to understand and maintain. At the end of the function, return the values as a tuple if needed.
```
bool match = false;
SERVER_TYPE server_type = SERVER_TYPE_MYSQL;
time_t time_stamp = 0;
...
...
return std::make_tuple(match, server_type, time_stamp);
```

Reply:
Addressed.